milabench/pack.py

```diff
@@ -244,6 +245,24 @@ def launch(self, script, args=(), voirargs=(), env={}):
         return process
 
 
+@contextlib.contextmanager
+def inject_in_requirements(input_files:list, constraints:list, dir:XPath, inject_mb:bool=False):
+    injected_in = dir / next(i for i in input_files if XPath(i).suffix in {".in", ".txt"})
+    _content = injected_in.read_text()
+    try:
+        if inject_mb:
+            from importlib.metadata import version
+            injected_in.append_lines(f"milabench=={version('milabench')}")
```

Comment:
I'm not entirely sure we want to do this. It's useful to avoid requirements.txt files which make milabench crash but at the same time these files will need to be updated more often to following the dependencies update of the project

Reply:
I'm not sure. I think it is relatively unlikely for the dependencies to clash in practice, because several of them, e.g. numpy or blessed, are not going to be used by milabench in the benchmark's venv (they are used for reporting, by milabench's main process running in the main venv). Ultimately I think the benchmarks should only depend on voir and not milabench itself, but that adjustment may not be done in time.

My take would be that it is unnecessary.